lib/writeBinaryGltf.js

```diff
@@ -10,18 +10,25 @@ function writeBinaryGltf(gltf, outputPath, createDirectory, callback) {
     // Correct output path extension if necessary
     var outputExtension = path.extname(outputPath);
     if (outputExtension !== '.glb') {
-        outputPath = path.basename(outputPath, outputExtension) + '.glb';
+        outputPath = path.join(path.dirname(outputPath), path.basename(outputPath) + '.glb');
     }
     // Create the output directory if specified
     if (createDirectory) {
         outputPath = path.join(path.dirname(outputPath), 'output', path.basename(outputPath));
-        mkdirp.sync(path.dirname(outputPath));
     }
+    mkdirp.sync(path.dirname(outputPath));
 
-    var glb = getBinaryGltf(gltf, callback);

```

Comment:
This callback was throwing off callbacks that expect an error in the first argument, but instead got the header, scene, body instead.